onnxruntime/core/session/compile_api.cc

```diff
@@ -0,0 +1,242 @@
+  }
+
+  if (allocator == nullptr) {
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "Invalid allocator for output model buffer: allocator pointer is null");
+  }
+
+  OrtSessionOptions* session_options = model_compile_options->session_options.get();
+  session_options->value.ep_context_gen_options.output_model_buffer_ptr = output_model_data_ptr;
+  session_options->value.ep_context_gen_options.output_model_buffer_size_ptr = output_model_data_size_ptr;
+  session_options->value.ep_context_gen_options.output_model_buffer_allocator = allocator;
+  return nullptr;
+  API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(OrtCompileAPI::ModelCompilationOptions_SetEpContextEmbedMode,
+                    _In_ OrtModelCompilationOptions* ort_model_compile_options,
+                    bool embed_ep_context_in_model) {
+  API_IMPL_BEGIN
+  auto model_compile_options = reinterpret_cast<onnxruntime::ModelCompilationOptions*>(ort_model_compile_options);
+  OrtSessionOptions* session_options = model_compile_options->session_options.get();
+  ORT_C_API_RETURN_IF_ERROR(session_options->value.config_options.AddConfigEntry(kOrtSessionOptionEpContextEmbedMode,
+                                                                                 embed_ep_context_in_model ? "1" : "0"));
+  session_options->value.ep_context_gen_options.embed_ep_context_in_model = embed_ep_context_in_model;
+  return nullptr;
+  API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(OrtCompileAPI::CompileModel, _In_ const OrtEnv* env,
+                    _In_ const OrtModelCompilationOptions* ort_model_compile_options) {
+  API_IMPL_BEGIN
+  auto model_compile_options = reinterpret_cast<const onnxruntime::ModelCompilationOptions*>(ort_model_compile_options);
+  ORT_C_API_RETURN_IF_ERROR(model_compile_options->Check());
+
+  std::unique_ptr<onnxruntime::InferenceSession> session;
+  OrtStatus* status = nullptr;
+
+  ORT_TRY {
+    if (!model_compile_options->input_model_path.empty()) {
+      PathString input_model_path = ToPathString(model_compile_options->input_model_path);
+      ORT_API_RETURN_IF_ERROR(CreateSessionAndLoadModel(model_compile_options->session_options.get(), env,
+                                                        input_model_path.c_str(),
+                                                        nullptr, 0, session));
+    } else {
+      ORT_API_RETURN_IF_ERROR(CreateSessionAndLoadModel(model_compile_options->session_options.get(), env, nullptr,
+                                                        model_compile_options->input_model_data,
+                                                        model_compile_options->input_model_data_size, session));
+    }
+    ORT_API_RETURN_IF_ERROR(InitializeSession(model_compile_options->session_options.get(), *session));
+  }
+  ORT_CATCH(const std::exception& e) {
+    ORT_HANDLE_EXCEPTION([&]() {
+      status = OrtApis::CreateStatus(ORT_FAIL, e.what());
+    });
+  }
```

Comment:
Do we need this or is the generic API_IMPL_BEGIN/API_IMPL_END enough? if we were adding something to the error message in the catch I could see value in it, but we're not currently.

Reply:
Removed these try/catch.